core/client/ui/user-onboarding.js

```diff
@@ -0,0 +1,110 @@
+const ONBOARDING_LAST_STEP = 3;
+const keyboards = ['q', 'z', 's', 'd', 'up', 'down', 'right', 'left'];
+const isMobile = window.innerWidth < 480;
+
+const getDirectionFromKey = key => {
+  switch (key) {
+    case 'z':
+    case 'up':
+      return 'up';
+    case 's':
+    case 'down':
+      return 'down';
+    case 'q':
+    case 'left':
+      return 'left';
+    case 'd':
+    case 'right':
+      return 'right';
+    default:
+      return null;
+  }
+};
+
+
+const requestUserMedia = async () => {
+  const constraints = userStreams.getStreamConstraints(streamTypes.main);
+  const stream = await userStreams.requestUserMedia(constraints);
+  if (!stream) { lp.notif.error(`unable to get a valid stream`); return; }
+
+  Session.set('streamAccepted', true);
+
+  // We should stop the stream directly after asking permissions, since we just want to check if the user has granted permissions
+  userStreams.destroyStream(streamTypes.main);
+};
+
+const bindKeyboards = () => {
+  keyboards.forEach(key => {
+    hotkeys(key, { keyup: true }, event => {
+      if (event.repeat) return;
+
+      const learnedDirections = Session.get('learnedDirections') || [];
+
+      if (event.type === 'keydown') {
+        const direction = getDirectionFromKey(key);
+
+        if (!learnedDirections.includes(direction)) learnedDirections.push(direction);
+
+        Session.set('learnedDirections', learnedDirections);
+        Session.set('pressedKeyboard', key);
+      } else {
+        Session.set('pressedKeyboard', null);
+      }
+    });
+  });
+};
+
+const finishOnboarding = () => {
+  Meteor.users.update(Meteor.userId(), { $unset: { 'profile.guest': true } });
```

Comment:
I think we have a different definition of guest status. For us, the "guest" status corresponds to the fact of being able to use lemverse without having created an account. Once the account is created, we remove this "guest" status. By doing this you turn the user account into an uninvited one, which we want to avoid.

Maybe we should base on something else for the onboarding, for example have a session variable indicating the current onboarding step, if it exists and is greater than 0 we display the onboarding.

Then to trigger this screen on page load you would have to :
- Have the guest attribute on the user
- Don't have an attribute in the local storage like `"onboarding-done": true`

Then set `Session.set('onboarding-step', 1)`.

The advantage is that we could trigger onboarding again even on an existing user. What do you think about this?